pkg/polylog/polyzero/logger_test.go

```diff
@@ -0,0 +1,369 @@
+package polyzero_test
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/pkg/polylog"
+	"github.com/pokt-network/poktroll/pkg/polylog/polyzero"
+	"github.com/pokt-network/poktroll/testutil/testpolylog"
+)
+
+const polyzeroEventTypeName = "*polyzero.zerologEvent"
+
+var (
+	expectedErr  = fmt.Errorf("%d", 42)
+	expectedTime = time.Now()
+	// NB: #Timestamp() uses time.Now() internally. If the test is run around the
+	// rollover of a second, minute, or hour, the expected timestamp time may not
+	// match the actual time precisely enough. While this is still a possibility
+	// near the rollover of a day, this window occurs less frequently and is many
+	// multiples of the time it takes CI to run.
+	//
+	// TODO_CONSIDERATION: redesign the test helper to support regular expressions
+	// for the output expectation.
+	expectedTimestampDayPrecisionLayout = "2006-01-02T"
```

Comment:
What happened on January 2nd 2006!?

Reply:
Haha, this is just the "reference time", see the [time pkg docs](https://pkg.go.dev/time#pkg-constants). Basically, instead of specifying a format using placeholders like most other languages, you just format an actual (but specific) date/time and the library does the right thing.

Reply:
Fascinating. I'd say this deserves a #PUC